saito-analytics/src/runner.rs

```diff
@@ -0,0 +1,169 @@
+use saito_core::core::data::crypto::{generate_keys, generate_random_bytes, hash, verify_signature};
+use saito_core::core::data::golden_ticket::GoldenTicket;
+use saito_core::core::data::mempool::Mempool;
+use saito_core::core::data::network::Network;
+use saito_core::core::data::peer_collection::PeerCollection;
+use saito_core::core::data::storage::Storage;
+use saito_core::core::data::transaction::{Transaction, TransactionType};
+use saito_core::core::data::wallet::Wallet;
+use saito_core::core::mining_thread::MiningEvent;
+use saito_core::{lock_for_read, lock_for_write};
+use saito_core::core::data::configuration::{Configuration, PeerConfig, Server};
+use crate::config::TestConfiguration;
+
+//use crate::sutils::*;
+use crate::sutils::load_blocks_disk;
+
+fn print_type_of<T>(_: &T) {
+    println!("{}", std::any::type_name::<T>())
+}
+
+pub fn create_timestamp() -> Timestamp {
+    SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .unwrap()
+        .as_millis() as Timestamp
+}
+
+//struct to manage setup of chain
+pub struct ChainRunner {
+    pub mempool: Arc<RwLock<Mempool>>,
+    pub blockchain: Arc<RwLock<Blockchain>>,
+    pub wallet_lock: Arc<RwLock<Wallet>>,
+    pub latest_block_hash: SaitoHash,
+    pub network: Network,
+    pub storage: Storage,
+    pub peers: Arc<RwLock<PeerCollection>>,
+    pub sender_to_miner: Sender<MiningEvent>,
+    pub receiver_in_miner: Receiver<MiningEvent>,
+    pub configs: Arc<RwLock<dyn Configuration + Send + Sync>>,
+}
+
+impl ChainRunner {
+    pub fn new() -> Self {
+        let keys = generate_keys();
+        let wallet = Wallet::new(keys.1, keys.0);
+        let _public_key = wallet.public_key.clone();
+        let _private_key = wallet.private_key.clone();
+        let peers = Arc::new(RwLock::new(PeerCollection::new()));
+        let wallet_lock = Arc::new(RwLock::new(wallet));
+        let blockchain = Arc::new(RwLock::new(Blockchain::new(wallet_lock.clone())));
+        let mempool = Arc::new(RwLock::new(Mempool::new(wallet_lock.clone())));
+
+        let (sender_to_miner, receiver_in_miner) = tokio::sync::mpsc::channel(10);
+        let configs = Arc::new(RwLock::new(TestConfiguration {}));
+
+        Self {
+            wallet_lock: wallet_lock.clone(),
+            blockchain,
+            mempool,
+            latest_block_hash: [0; 32],
+            network: Network::new(
+                Box::new(TestIOHandler::new()),
+                peers.clone(),
+                wallet_lock.clone(),
+                configs.clone(),
+            ),
+            peers: peers.clone(),
+            storage: Storage::new(Box::new(TestIOHandler::new())),
+            sender_to_miner: sender_to_miner.clone(),
+            receiver_in_miner,
+            configs,
+        }
+    }
+
+    pub fn get_mempool(&self) -> Arc<RwLock<Mempool>> {
+        return self.mempool.clone();
+    }
+
+    pub fn get_wallet_lock(&self) -> Arc<RwLock<Wallet>> {
```

Comment:
it's better to not have getters/setters if possible in rust

Reply:
deleted, its in testmanager too